app/addons/documents/rev-browser/rev-browser.components.react.jsx

```diff
@@ -0,0 +1,446 @@
+              >
+                <i className="icon-columns" /> Diff
+              </Button>
+              <Button
+                style={{width: '120px'}}
+                className={isDiffViewEnabled ? '' : 'active'}
+                onClick={this.toggleDiffView.bind(this, false)}
+              >
+                <i className="icon-file-text" /> Document
+              </Button>
+            </ButtonGroup>
+          </div>
+
+          {isDiffViewEnabled ?
+            <RevisionDiffArea ours={ours} theirs={theirs} /> :
+            <SplitScreenArea ours={ours} theirs={theirs} /> }
+        </div>
+      );
+    }
+  };
+
+
+  class SplitScreenArea extends React.Component {
+
+    constructor (props) {
+      super(props);
+    }
+
+    componentDidUpdate () {
+      this.hightlightAfterRender();
+    }
+
+    componentDidMount () {
+      this.hightlightAfterRender();
+    }
+
+    hightlightAfterRender () {
+      const format = (input) => { return JSON.stringify(input, null, '  '); };
+
+      const jsmode = new JavaScriptMode();
+      const left = ReactDOM.findDOMNode(this.refs.revLeftOurs);
+      const right = ReactDOM.findDOMNode(this.refs.revRightTheirs);
+
+      const leftRes = highlight.render(format(this.props.ours), jsmode, theme, 0, true);
+      left.innerHTML = leftRes.html;
+      const rightRes = highlight.render(format(this.props.theirs), jsmode, theme, 0, true);
+      right.innerHTML = rightRes.html;
+    }
+
+    render () {
+      const {ours, theirs} = this.props;
+
+      if (!ours || !theirs) {
+        return <div></div>;
+      }
+
+      return (
+        <div className="revision-split-area" style={{padding: '20px 15px', display: 'flex', color: '#fff'}}>
+          <div data-id="ours" style={{width: '50%'}}>
+            <div style={{marginBottom: '20px'}}>{ours._rev} (Server-Selected Rev)</div>
+            <pre ref="revLeftOurs"></pre>
+          </div>
+
+          <div data-id="theirs" style={{width: '50%'}}>
+            <div style={{marginBottom: '20px'}}>{theirs._rev}</div>
+            <pre ref="revRightTheirs"></pre>
+          </div>
+        </div>
+      );
+    }
+  }
+
+  const RevisionDiffArea = ({ours, theirs}) => {
+    if (!ours || !theirs) {
+      return <div></div>;
+    }
+
+    const delta = jdp.diff(ours, theirs);
+    const html = jdpformatters.format(delta, ours);
+
+    return (
+      <div className="revision-diff-area">
+        <div
+          style={{marginTop: '30px'}}
+          dangerouslySetInnerHTML={{__html: html}}
+        >
+        </div>
+      </div>
+    );
+  };
+  RevisionDiffArea.propTypes = {
+    ours: React.PropTypes.object,
+    theirs: React.PropTypes.object,
+    currentRev: React.PropTypes.string
+  };
+
+
+  const ConflictingRevisionsDropDown = ({options, selected, onRevisionClick, onBackwardClick, onForwardClick}) => {
+    return (
+      <div style={{maxWidth: '370px', display: 'flex', marginLeft: '-17px'}}>
```

Comment:
aw, good point. as i read this i woudl have had a point for our dev COP today: that i feel unsure about inline styles now, meh, next cop